attestation-agent/kbs_protocol/src/api.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::Result;
+use async_trait::async_trait;
+pub use resource_uri::ResourceUri;
+
+use crate::{keypair::TeeKeyPair, token_provider::Token};
+
+#[async_trait]
+pub trait PassportClientCapabilities {
+    async fn get_token(&mut self) -> Result<(Token, TeeKeyPair)>;
```

Comment:
Not related to the questions: let me delete the trait `PassportClientCapabilities` and just have a function `get_token()`, which is only to show that the client "can provide a token".

So the questions:

> So the token will always be linked to the TeeKeyPair and there is no option to provide any other user data? Will that cover all use cases?

Yes. This is a `KbsClient<Box<dyn EvidenceProvider>>` and it can generate a key pair and a token which certifies the token. RCAR  handshake (Remote attestation, evidence exchange ...) will be performed underlying

The key pair is bonded to a token. As KBS protocol defines the responsed data should use the key certified by the token to unwrap https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#response, they should be given together

Reply:
I guess it's just following the protocol that we've settled on, so this should be fine. I think there might be some other possible use cases but we can worry about that later.